crates/sui-rpc-api/proto/sui/rpc/v2alpha/object.proto

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package sui.rpc.v2alpha;
+
+import "sui/rpc/v2alpha/bcs.proto";
+import "sui/rpc/v2alpha/owner.proto";
+
+// An object on the Sui blockchain.
+message Object {
+  // This Object serialized as BCS.
+  optional Bcs bcs = 1;
+
+  // `ObjectId` for this object.
+  optional string object_id = 2;
+
+  // Version of the object.
+  optional uint64 version = 3;
+
+  // The digest of this Object.
+  optional string digest = 4;
+
+  // Owner of the object.
+  optional Owner owner = 5;
+
+  // The type of this object.
+  optional string object_type = 6;
+
+  // DEPRECATED this field is no longer used to determine whether a tx can transfer this
+  // object. Instead, it is always calculated from the objects type when loaded in execution.
+  //
+  // Only set for Move structs
```

Comment:
Thoughts on introducing a `MoveObject` and `MovePackage` type, and using `oneof` here?

Reply:
I did have this before and we can add that back. I did shift fully from oneof to flattening in most places but maybe here going back to one of would be ok. My thinking on not doing that was mostly on the awkwardness of having a type that only has a part of the information for those things (version, objectid, etc) wouldn't be present there unless we duplicated it and I wanted to avoid that

Reply:
I see, makes sense -- the only reason I raised it was for ease of distinguishing between objects and packages (IIUC, currently you would have to pick a field that only exists for one or the other and check whether it exists). This could be solved with an enum just to signify that though.